topostats/tracing/dnatracing.py

```diff
@@ -154,6 +158,86 @@ def trace_dna(self):
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+
+        # Get the heights of each pixel in the ordered trace from the gaussian filtered image
+        # the pixel coordinates are stored in the ordered trace list
+        self.trace_heights = list(self.gauss_image[self.ordered_trace[:, 0], self.ordered_trace[:, 1]])
+
+    def get_trace_cumulative_distances(self) -> None:
+        """
+        Populate a `dnaTrace` instance's `self.trace_distances` list with the cumulative distances of
+        each pixel in the `self.ordered_trace` list.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+
+        # Get the cumulative distances of each pixel in the ordered trace from the gaussian filtered image
+        # the pixel coordinates are stored in the ordered trace list.
+        self.trace_cumulative_distances = self.coord_dist(
+            coordinates=self.ordered_trace, px_to_nm=self.pixel_to_nm_scaling
+        )
+
+    @staticmethod
+    def coord_dist(coordinates: np.ndarray, px_to_nm: float) -> np.ndarray:
+        """
+        Returns a list of the cumulative real distances between each pixel in a trace.
+
+        Take a Nx2 numpy array of (grid adjacent) coordinates and produce a list of cumulative distances in
+        nanometres, travelling from pixel to pixel. 1D example: coordinates: [0, 0], [0, 1], [1, 1], [2, 2] cumulative
+        distances: [0, 1, 2, 3.4142]. Counts diagonal connections as 1.4142 distance. Converts distances from
+        pixels to nanometres using px_to_nm scaling factor.
+
+        Note that the pixels have to be adjacent.
+
+        Parameters
+        ----------
+        coords: np.ndarray
+            A Nx2 integer array of coordinates of the pixels of a trace from a binary trace image.
+        px_to_nm: float
+            Pixel to nanometre scaling factor to allow for real length measurements of distances rather
+            than pixels.
+
+        Returns
+        -------
+        list
+            List of length N containing the cumulative sum of distances (0 at the first entry,
+            full molecule length at the last entry)
+        """
+
+        cumulative_distance_list = [0]
+        distance = 0
+        for i in range(len(coordinates) - 1):
+            # If the pixels are diagonally touching, add sqrt 2
+            if abs(coordinates[i] - coordinates[i + 1]).sum() == 2:
+                distance += 1.41421356237  # Saves a sqrt call
+            else:
+                distance += 1.0
+            cumulative_distance_list.append(distance)
+        # Convert to array first to allow for multiplication by px_to_nm
+        # Then convert back to list since JSON cannot save as numpy array
+        # Better to do the conversion here than have to do it in the
+        # JSON saving script.
+        return list(np.asarray(cumulative_distance_list) * px_to_nm)
```

Comment:
No need to convert to an array as can just use list comprehension
```suggestion
        return [distance * px_to_nm for distance in cumulative_distance_list]
```

Reply:
Also does this make redundant the `recursive_numpy_array_to_list()` function if as noted `Better to do the conversion here than have to do it in the JSON saving script`?